packages/jest-cli/src/search_source.js

```diff
@@ -249,18 +249,25 @@ export default class SearchSource {
       changedFilesPromise,
     );
 
-    if (globalConfig.filter) {
-      let tests = searchResult.tests;
+    const filterPath = globalConfig.filter;
+
+    if (filterPath) {
+      const tests = searchResult.tests;
 
       // $FlowFixMe: dynamic require.
-      const filter = require(globalConfig.filter);
+      const filter = require(filterPath);
       const filtered = await filter(tests.map(test => test.path));
 
-      if (Array.isArray(filtered)) {
-        tests = tests.filter(test => filtered.includes(test.path));
+      if (!Array.isArray(filtered)) {
+        throw new Error(
+          `Filter ${filterPath} did not return a valid test list`,
+        );
       }
 
-      searchResult.tests = tests;
+      // $FlowFixMe: Object.assign with empty object causes troubles to Flow.
```

Comment:
If I do:

```javascript
{
  ...searchResult,
  tests: <whatever>,
}
```

It works. But with `Object.assign` it does not. And they're actually the same... but we don't support spread on transpilation, so I need the `$FlowFixMe` 🤷‍♂️